File: library/src/main/java/com/yongchun/library/view/ImagePreviewFragment.java
package com.yongchun.library.view;

import android.graphics.Bitmap;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.GestureDetector;
import android.view.LayoutInflater;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.bumptech.glide.request.animation.GlideAnimation;
import com.bumptech.glide.request.target.SimpleTarget;
import com.github.chrisbanes.photoview.OnPhotoTapListener;
import com.github.chrisbanes.photoview.PhotoView;
import com.yongchun.library.R;

import java.io.File;


public class ImagePreviewFragment extends Fragment {
    private PhotoView photo_view;

    public static final String PATH = "path";

    public static ImagePreviewFragment getInstance(String path) {
        ImagePreviewFragment fragment = new ImagePreviewFragment();
        Bundle bundle = new Bundle();
        bundle.putString(PATH, path);
        fragment.setArguments(bundle);
        return fragment;
    }

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View contentView = inflater.inflate(R.layout.fragment_image_preview, container, false);
        initView(container, contentView);
        return contentView;
    }

    private void initView(ViewGroup container, View contentView) {
        photo_view = (PhotoView) contentView.findViewById(R.id.photo_view);


        Glide.with(container.getContext())
                .load(new File(getArguments().getString(PATH)))
                .asBitmap()
                .into(new SimpleTarget<Bitmap>(480, 800) {
                    @Override
                    public void onResourceReady(Bitmap resource, GlideAnimation<? super Bitmap> glideAnimation) {
                        photo_view.setImageBitmap(resource);
                    }
                });
        photo_view.setOnPhotoTapListener(new OnPhotoTapListener() {
            @Override
            public void onPhotoTap(ImageView view, float x, float y) {
                ImagePreviewActivity activity = (ImagePreviewActivity) getActivity();
                activity.switchBarVisibility();
            }
        });

    }


}

File: imagepicker/src/main/java/com/weimu/library/ui/ImagePreviewFragment.kt
package com.weimu.library.ui

import android.os.Bundle
import com.weimu.library.R
import com.weimu.universalview.core.fragment.BaseFragment
import com.weimu.universalview.ktx.load
import com.weimu.universalview.ktx.load4CenterCrop
import kotlinx.android.synthetic.main.fragment_image_preview.*


internal class ImagePreviewFragment : BaseFragment() {
    override fun afterViewAttachBaseViewAction(savedInstanceState: Bundle?) {}

    override fun beforeViewAttachBaseViewAction(savedInstanceState: Bundle?) {}

    override fun getLayoutResID(): Int = R.layout.fragment_image_preview

    override fun afterViewAttach(savedInstanceState: Bundle?) {
        photo_view.load("${arguments!!.getString(PATH)}")
        photo_view.setOnPhotoTapListener { view, x, y ->
            val activity = activity as ImagePreviewActivity
            activity.isShowBar = !activity.isShowBar
        }
    }


    companion object {

        val PATH = "path"

        fun getInstance(path: String): ImagePreviewFragment {
            val fragment = ImagePreviewFragment()
            val bundle = Bundle()
            bundle.putString(PATH, path)
            fragment.arguments = bundle
            return fragment
        }
    }


}
